extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.ts

```diff
@@ -156,18 +156,44 @@ export default class StaticWadoClient extends api.DICOMwebClient {
    *
    * @param {*} desired
    * @param {*} actual
+   * @param {*} fuzzyMatching - if true, then do a sub-string match on the values
    * @returns true if the values match
    */
-  compareValues(desired, actual) {
+  compareValues(desired, actual, fuzzyMatching) {
     if (Array.isArray(desired)) {
-      return desired.find(item => this.compareValues(item, actual));
+      return desired.find(item => this.compareValues(item, actual, fuzzyMatching));
     }
     if (Array.isArray(actual)) {
-      return actual.find(actualItem => this.compareValues(desired, actualItem));
+      return actual.find(actualItem => this.compareValues(desired, actualItem, fuzzyMatching));
     }
     if (actual?.Alphabetic) {
       actual = actual.Alphabetic;
     }
+
+    if (fuzzyMatching && typeof actual === 'string' && typeof desired === 'string') {
+      const normalizeValue = str => {
+        if (str.startsWith('*')) {
```

Comment:
If you supported wildcard matching here too, with prefix/suffix, then you can support fuzzy matching for names, and wildcard start/ends with for other fields.  I would suggest changing the support of supportsWildcard to `suffix`, to automatically have the wildcard added only at the end, although doing a full contains works as well for client side matching.

Reply:
I think this implementation is sufficient for now. We can dedicate more time to implementing proper server-side filtering logic for static wado using lamdas or something similar then we can avoid all of this in the frontend.